rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
```

Comment:
```suggestion
 * \param[out] directory The C string pointer at which to write the directory path.
```
We're not reading `directory`, just using it as an output.

Reply:
fixed in 0705dd3a57a3e36594ac9a2d054a1b54a715fef8